torch/_dynamo/variables/functions.py

```diff
@@ -601,3 +601,40 @@ def call_function(
                 f"CollectiveFunctionRewriteVariable can't support async_op=True for {self.orig_fn}"
             )
         return super().call_function(tx, args, kwargs)
+
+
+class FunctoolsPartialVariable(VariableTracker):
+    def __init__(self, func, args, keywords, **kwargs):
+        super().__init__(**kwargs)
+        # This must be a callable. It is sound to expand this list to other callables.
+        assert isinstance(
+            func,
+            (
+                variables.UserFunctionVariable,
+                variables.NestedUserFunctionVariable,
+                variables.TorchVariable,
+                variables.NNModuleVariable,
```

Comment:
I don't understand, why do you need the isinstance assert here? If you are testing if the variable is callable, you should faithfully simulate what `callable()` semantics are

Reply:
Also, this isn't an assert, you can trigger this from user code